packages/js/components/src/experimental-select-control/menu.tsx

```diff
@@ -16,9 +17,48 @@ type MenuProps = {
 };
 
 export const Menu = ( { children, getMenuProps, isOpen }: MenuProps ) => {
+	const [ boundingRect, setBoundingRect ] = useState< DOMRect >();
+	const selectControlMenuRef = useRef< HTMLDivElement >( null );
+
+	useEffect( () => {
+		if (
+			selectControlMenuRef.current &&
+			selectControlMenuRef.current?.parentElement
+		) {
+			setBoundingRect(
+				selectControlMenuRef.current?.parentElement.getBoundingClientRect()
+			);
+		}
+	}, [ selectControlMenuRef.current ] );
+
+	/* eslint-disable jsx-a11y/no-noninteractive-element-interactions, jsx-a11y/click-events-have-key-events */
+	const childrenPopoverMarkup = (
```

Comment:
Having component markup this large inside another component feels a bit like a code smell.  Should we move it inside the render below or to a separate popover component file?

Reply:
Thanks Josh I moved it down to the render return, as pushing it to a different component felled a bit like over kill as it practically required all the same `props`: https://github.com/woocommerce/woocommerce/pull/34967/commits/8b4c672c2615805a1caf1a5a84017a3593232bbe